src/QMCWaveFunctions/TrialWaveFunction.h

```diff
@@ -167,9 +176,20 @@ class TrialWaveFunction
    */
   void reportStatus(std::ostream& os);
 
+  void attachTWFFastDerivWrapper(std::unique_ptr<TWFFastDerivWrapper> twf_fdw) noexcept;
+
+
   /** Initialize a TWF wrapper for fast derivative evaluation
    */
   void initializeTWFFastDerivWrapper(const ParticleSet& P, TWFFastDerivWrapper& twf) const;
+
+
+  /** Non-owning view of the TWF-owned wrapper)
+    */
+  TWFFastDerivWrapper* getTWFFastDerivWrapper() noexcept { return twf_fastderiv_.get(); }
+  const TWFFastDerivWrapper* getTWFFastDerivWrapper() const noexcept { return twf_fastderiv_.get(); }
```

Comment:
Strongly against raw pointers. They seem to be used as reference and then please just use reference type. Since they are not used by this PR, please remove them in this PR and introduce the reference version when needed.